### ISSUE TITLE
Google Suchergebnis-Vorschau: Eingabe der Zeichen zählen (SEO)

### ISSUE THREAD

**Issue by minstyle** [contao/contao#3232]
Bei Eingabe des _Seitentitels_ und der _Beschreibung der Seite_ wäre es schön, wenn in der Google Suchergebnis-Vorschau auch die Anzahl der eingegebenen Zeichen angezeigt werden, um die Textlängen besser kontrollieren zu können.

![image](https://user-images.githubusercontent.com/2649562/127625739-8e1a7192-601b-4498-8d20-29ee968b4b12.png)

Sind bspw. die Angaben für `Meta Title` und `Meta Description` zu lang, so werden die Texte in Google nicht vollständig angezeigt und diese mit Auslassungszeichen dargestellt.

**Comment by leofeyer** [comment]
Die Vorschau stellt die Texte genauso wie Google dar und fügt ebenfalls Auslassungszeichen ein, wenn die Eingabe zu lang ist. Ist es wirklich notwendig, zusätzlich noch die Anzahl der Zeichen anzuzeigen?

**Comment by m-vo** [comment]
If at all I would add this information to the `description` field as an eval option for the widget (like `showChars` or something), not the `serpPreview`. There it might in general be useful as a feedback for editors when setting min/max limits.

**Comment by bekanntmacher** [comment]
Google geht nach Pixel nicht nach Zeichenzahl weil ein `i` ungleich viele Pixel wie ein `w` benötigt. 

Titel: 580 Pixel (Desktop) beziehungsweise 920 Pixel (Mobile)
Description: 990 Pixel (Desktop) beziehungsweise 1.300 Pixel (Mobile)

Die perfekte Lösung finde ich diese https://app.sistrix.com/de/serp-snippet-generator.

**Comment by fritzmg** [comment]
Any in any case what Google displays or does not display is never constant anyway. Most of the time it will not even show you the meta description.

**Comment by bekanntmacher** [comment]
Ist deine Aussage irgendwie belegbar?

**Comment by fritzmg** [comment]
Which one? That Google search result content snippets change in length an appearance or that it will not show you the meta description? You can test the latter with a Google search yourself 😉. Google will show you relevant snippets of the actual content you are searching for, rather than the meta description.

**Comment by bekanntmacher** [comment]
Ich habe gerade in Google mit Contao gesucht und es zeigt mir genau die Beschreibung, die ich im Quelltext finde.

**Comment by fritzmg** [comment]
If you search for [contao gdpr](https://www.google.com/search?q=contao+gdpr) the result will be this:

<img src="https://github.com/contao/contao/assets/4970961/d00c1978-8d40-429f-b9f1-90c16b02d0b0" width="574">

**Comment by bekanntmacher** [comment]
Ja das ist auch total okay so. 

Würde eine genügend gewichtige Unterseite zu gdpr existieren, würde Google diese Unterseite anzeigen mit deren Description. Existiert keine Unterseite zum entsprechenden Begriff, nimmt er die am ehesten passende.

**Comment by fritzmg** [comment]
> Würde eine genügend gewichtige Unterseite zu gdpr existieren, würde Google diese Unterseite anzeigen mit deren Description.

No, even in that case Google might show snippets of the content, since that is what you are searching for. In any case, it is completely up to Google what it shows you. You can never be sure what Google will show and how much.

**Comment by bekanntmacher** [comment]
> No, even in that case Google might show snippets of the content, since that is what you are searching for.

Kannst du dieses Argument irgendwie belegen?

Google kannst du nicht zwingen, das ist klar. Aber ein optimiertes Snippet steigert die Klickrate. Und ein Snippet mit der passenden Länge maximiert die Chance, dass er es nimmt.

**Comment by fritzmg** [comment]
> Kannst du dieses Argument irgendwie belegen?

[contao partners](https://www.google.com/search?q=contao+partners)

<img src="https://github.com/contao/contao/assets/4970961/f2bc1a42-533c-4a7d-a188-9bdbba11f6d7" width="606">

It does not show the meta description of the [Contao partners](https://contao.org/en/partners) page. As I said, it is completely up to Google what it will show you. There is no guarantee that Google will show the meta description for your search query at all. Only when it thinks it is helpful to you - the person wanting to find specific content. And arguably in this example the actual meta description _would_ be more helpful than what Google decided to show. But alas, it just does not show it.

**Comment by bekanntmacher** [comment]
Bei deinem Beispiel haut Google voll daneben, das ist so. Bei mir nimmt er die definierte Descrition. Und ja, Google lässt sich in der Regel nichts vorschreiben (ausser vom Gesetz). 

Die Hauptgründe für eine Ablehnung durch Google sind in der Regel folgende:

1. Es gibt keine Description.
2. Die Description ist zu lang oder viel zu kurz.
3. Die Description passt nicht zum Inhalt der Seite.
4. Die Description passt nicht zum Suchbegriff, die Seite aber schon.
6. Die Description ist sehr schlecht formuliert.
7. Die Seite verändert sich dynamisch.

Ich vermute, dass Google in Zukunft sowohl die Description als auch den Titel immer selbst formulieren wird (KI).
Im Moment ist es aber so, dass die Chancen gut stehen, dass er eine saubere Description (Es gibt natürlich ausser der Länge noch andere Punkte die beachtet werden müssen, aber das ist ein anderes Thema) übernimmt. Und dann, das haben wir an Beispielen gemessen, steigt die Klickrate stark an.

Bei deinem Beispiel kommt vermutlich 6. zum tragen. Oder Google versteht die Seite nicht. Dann wäre es 3. 

Ich wollte einfach folgendes einbringen und das ist so auch nachvollziehbar: Da die Zeichenbreite variiert, macht es keinen Sinn, einen Zeichenzähler einzubauen. Wenn, dann wäre eine Lösung wie Sistrix das Richtige. Ob man das in Contao einbauen will, ist eine andere Frage. 

Schöne Grüsse und weiterhin frohes Schaffen

**Comment by fritzmg** [comment]
> Da die Zeichenbreite variiert, macht es keinen Sinn, einen Zeichenzähler einzubauen.

I agree. But my main point is that catering the meta description specifically "for Google" is the wrong approach in general.
